api/strategies/openidStrategy.js

```diff
@@ -0,0 +1,78 @@
+const passport = require('passport');
+const express = require('express');
+const jwt = require('jsonwebtoken');
+const { Issuer, Strategy: OpenIDStrategy } = require('openid-client');
+const config = require('../../config/loader');
+const domains = config.domains;
+
+const User = require('../models/User');
+
+Issuer.discover(process.env.OPENID_ISSUER)
+  .then(issuer => {
+    const client = new issuer.Client({
+      client_id: process.env.OPENID_CLIENT_ID,
+      client_secret: process.env.OPENID_CLIENT_SECRET,
+      redirect_uris: [domains.server + process.env.OPENID_CALLBACK_URL]
+    });
+
+    const openidLogin = new OpenIDStrategy(
+      {
+        client,
+        params: {
+          scope: process.env.OPENID_SCOPE
+        }
+      },
+      async (tokenset, userinfo, done) => {
+        try {
+          let user = await User.findOne({ email: userinfo.email });
```

Comment:
Just confirming for security as I don't know much about its inner workings, can OpenID be trusted to only have verified email accounts? I'm assuming so but just wanted to double check